sdk/multisig/src/actions/common/base.ts

```diff
@@ -0,0 +1,289 @@
+import {
+  Connection,
+  Keypair,
+  PublicKey,
+  TransactionInstruction,
+  TransactionMessage,
+  TransactionSignature,
+  VersionedTransaction,
+} from "@solana/web3.js";
+import {
+  BaseBuilderArgs,
+  BuildResult,
+  BuildTransactionSettings,
+  SendSettings,
+} from "./types";
+
+export abstract class BaseBuilder<
+  T extends BuildResult,
+  U extends BaseBuilderArgs = BaseBuilderArgs
+> {
+  public createKey?: Keypair;
+  protected connection: Connection;
+  protected instructions: TransactionInstruction[] = [];
+  protected creator: PublicKey = PublicKey.default;
+  protected buildPromise: Promise<void>;
+  protected args: Omit<U, keyof BaseBuilderArgs>;
+  private built: boolean = false;
+  // Use this as an indicator to clear all instructions?
+  private sent: boolean = false;
```

Comment:
You may have already considered this, but if you want methods and properties to be "truly" private, meaning they can't be accessed with trickery, I recommend using the native JS private properties `#send: boolean = false.`

In case you're curious: [MDN: Private Properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties)